src/main/1.3/scala/com/holdenkarau/spark/testing/TestInputStream.scala

```diff
@@ -43,9 +42,11 @@ class TestInputStream[T: ClassTag](@transient var sc: SparkContext,
     logInfo("Computing RDD for time " + validTime)
     val index = ((validTime - ourZeroTime) / slideDuration - 1).toInt
     val selectedInput = if (index < input.size) input(index) else Seq[T]()
+    // TODO what if the number of batches doesn't equal size of input list !!
+    // for example => if input.size = 10, numOfBatches = 2 (Should we return 5 at every call !!)
 
     // lets us test cases where RDDs are not created
-    if (selectedInput == null) {
+    if (selectedInput == null) { //TODO will never be null !!
```

Comment:
I think it will always equals some value as it equals:
`val selectedInput = if (index < input.size) input(index) else Seq[T]()`

does that mean that the user can add null value explicitly ? and this means that this batch will be empty ?